src/overlays/actors/ovl_En_Famos/z_en_famos.h

```diff
@@ -2,18 +2,60 @@
 #define Z_EN_FAMOS_H
 
 #include "global.h"
+#include "objects/object_famos/object_famos.h"
 
 struct EnFamos;
 
 typedef void (*EnFamosActionFunc)(struct EnFamos*, GlobalContext*);
 
+typedef struct EnFamosRock {
+    /* 00 */ Vec3f pos;
+    /* 0C */ Vec3f velocity;
+    /* 18 */ Vec3s rotation;
+    /* 20 */ f32   scale;
+} EnFamosRock; // size = 0x24
+
+#define GET_FAMOS_PATH(thisx) (thisx->params)
+
+// params: [this->actor.shape.rot.x] is used to set agro distance
+
+// stateTimer gets reused:
+//   after spotting player, counts frames until first attack (8)
+//   after lost agro, measures frames looking around before returning to home (60)
+//   before smash attack, counts frames before disabling emblem colider (4)
+//   after smash attack collision, counts frames before rising (3)
+//   after smash attack finish, counts frames rising above ground before next attempt (20)
+
 typedef struct EnFamos {
-    /* 0x0000 */ Actor actor;
-    /* 0x0144 */ char unk_144[0x8C];
-    /* 0x01D0 */ EnFamosActionFunc actionFunc;
-    /* 0x01D4 */ char unk_1D4[0x440];
+    /* 0x000 */ Actor actor;
+    /* 0x144 */ SkelAnime skelAnime;
+    /* 0x188 */ Vec3s jointTable[FAMOS_LIMB_MAX];
+    /* 0x1AC */ Vec3s morphTable[FAMOS_LIMB_MAX];
+    /* 0x1D0 */ EnFamosActionFunc actionFunc;
+    /* 0x1D4 */ u8 animatedMaterialIndex;
+    /* 0x1D5 */ u8 stableRotation;
+    /* 0x1D6 */ u8 pathNodeCount;
+    /* 0x1D7 */ u8 currentPathNode;
+    /* 0x1D8 */ u8 isCalm; // is NOT aware of player
+    /* 0x1DA */ s16 hoverClk; // 30->0, repeat, for height adjustment when hovering
```

Comment:
Not clear what this is supposed to mean.

Reply:
Timer (count down), starts at 30 counts down once per frame to zero then resets to 30, used by the trig functions to identify what height adjustment Famos should have as it bobs up and down on a sin wave.

Reply:
I don't think you need to explain what values it takes, someone can read the code to find that out. And what does "Clk" mean?

Reply:
Sorry Clk is common abbreviation of "Clock" in hardware, wrong project.